File: agroguardian_app_android/app/src/main/java/com/example/agroguardianappandroid/data/models/Telemetry.kt
package com.example.agroguardianappandroid.data.models

import androidx.compose.ui.graphics.Color
import com.google.gson.annotations.SerializedName

data class Telemetry(
    @SerializedName("device_name")
    val deviceName: String,

    val temperature: Double,
    val humidity: Double,

    @SerializedName("soil_moisture")
    val soilMoisture: Double,

    val timestamp: String
) {
    fun temperatureColor(): Color {
        return when {
            temperature < 20 -> Color.Red
            temperature < 25 -> Color(0xFFF09F26)
            else -> Color.Green
        }
    }

    fun humidityColor(): Color {
        return when {
            humidity < 60 -> Color.Red
            humidity > 70 -> Color(0xFFF09F26)
            else -> Color.Green
        }
    }

    fun soilMoistureColor(): Color {
        return when {
            soilMoisture < 15 -> Color.Red
            soilMoisture > 60 -> Color(0xFFF09F26)
            else -> Color.Green
        }
    }
}

File: agroguardian_app_ios/agroguardian_app_ios_watch Watch App/Models/Telemetry.swift
//
//  Telemetry.swift
//  agroguardian_app_ios
//
//  Created by Eren on 8.05.2025.
//

import SwiftUICore

struct Telemetry: Codable {
    let deviceName: String
    let temperature: Double
    let humidity: Double
    let soilMoisture: Double
    let timestamp: String

    private enum CodingKeys: String, CodingKey {
        case deviceName = "device_name"
        case temperature, humidity, soilMoisture = "soil_moisture", timestamp
    }
    
    func temperatureColor() -> Color {
        if temperature < 20 {
            return .red
        } else if temperature < 25 {
            return .yellow
        } else {
            return .green
        }
    }
    
    func humidityColor() -> Color {
        if humidity < 60 {
            return .red
        } else if humidity > 70 {
            return .yellow
        } else {
            return .green
        }
    }
    
    func soilMoistureColor() -> Color {
        if soilMoisture < 15 {
            return .red
        } else if soilMoisture > 60 {
            return .yellow
        } else {
            return .green
        }
    }
}
